runtime/parachains/src/paras_inherent.rs

```diff
@@ -295,15 +520,262 @@ pub mod pallet {
 			// And track that we've finished processing the inherent for this block.
 			Included::<T>::set(Some(()));
 
-			Ok(Some(
-				MINIMAL_INCLUSION_INHERENT_WEIGHT +
-					(backed_candidates_len * BACKED_CANDIDATE_WEIGHT),
-			)
+			Ok(Some(paras_inherent_total_weight::<T>(
+				backed_candidates_len,
+				bitfields_len,
+				disputes_len,
+			))
 			.into())
 		}
 	}
 }
 
+macro_rules! ensure2 {
+	($condition:expr, $err:expr, $action:ident $(, $alt:expr)? $(,)?) => {
+		let condition = $condition;
+		if !condition {
+			if $action {
+				ensure!(condition, $err);
+			} else {
+				$($alt)?
```

Comment:
Just to clarify, what is the expected behaviour if there is no alt?

A little comment above this function explaining the expected behaviour might be nice

Reply:
If there is no `alt`, it will do nothing, as in the `else {}`-block will be empty.